libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -368,18 +368,25 @@ public void declareField(Parser<Value, Context> p, ParseField parseField, ValueT
         for (String fieldValue : parseField.getAllNamesIncludedDeprecated()) {
 
             if (RestApiVersion.minimumSupported().matches(parseField.getForRestApiVersion())) {
-                fieldParserMap.computeIfAbsent(RestApiVersion.minimumSupported(), (v)-> new HashMap<>())
+                FieldParser previousValue = fieldParserMap.computeIfAbsent(RestApiVersion.minimumSupported(), (v) -> new HashMap<>())
                     .putIfAbsent(fieldValue, fieldParser);
+                validateIfFieldAlreadyRegistered(previousValue, fieldValue);
             }
             if (RestApiVersion.current().matches(parseField.getForRestApiVersion())) {
-                fieldParserMap.computeIfAbsent(RestApiVersion.current(), (v)-> new HashMap<>())
+                FieldParser previousValue = fieldParserMap.computeIfAbsent(RestApiVersion.current(), (v)-> new HashMap<>())
                     .putIfAbsent(fieldValue, fieldParser);
-
+                validateIfFieldAlreadyRegistered(previousValue, fieldValue);
```

Comment:
I *think* this'd be a little easier to read if it were inlined. Having it like this names the action which is nice, but checking if the previous value isn't null is a fairly normal patter. I think.

It may be a little easier to read if the `computeIfAbsent` were on its own line and the `putIfAbsent` were on the next. :shrug: At this point I'm not sure any of those would be better. But when I was reading this I went "*huh*, is that right? it took me a few minutes to put it all in order."

Reply:
agree. Will inline the validation method and split the compute/putIfabsent calls